core/src/test/scala/unit/kafka/admin/LeaderElectionCommandTest.scala

```diff
@@ -0,0 +1,359 @@
+        assertTrue(e.getMessage.contains("election-type"))
+    }
+  }
+
+  @Test
+  def testMissingTopicPartitionSelection(): Unit = {
+    try {
+      LeaderElectionCommand.main(
+        Array(
+          "--bootstrap-server", bootstrapServers(servers),
+          "--election-type", "preferrred"
+        )
+      )
+      fail()
+    } catch {
+      case e: Throwable =>
+        assertTrue(e.getMessage.startsWith("Missing required option(s)"))
+        assertTrue(e.getMessage.contains("all-topic-partitions"))
+    }
+  }
+
+  @Test
+  def testInvalidBroker(): Unit = {
+    try {
+      LeaderElectionCommand.run(
+        Array(
+          "--bootstrap-server", "example.com:1234",
+          "--election-type", "unclean",
+          "--all-topic-partitions"
+        ),
+        1.seconds
+      )
+      fail()
+    } catch {
+      case e: AdminCommandFailedException =>
+        assertTrue(e.getCause.isInstanceOf[TimeoutException])
+    }
+  }
+}
+
+object LeaderElectionCommandTest {
+  def createConfig(servers: Seq[KafkaServer]): Map[String, Object] = {
+    Map(
+      AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG -> bootstrapServers(servers),
+      AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG -> "20000"
+    )
+  }
+
+  def bootstrapServers(servers: Seq[KafkaServer]): String = {
+    servers.map { server =>
+      val port = server.socketServer.boundPort(ListenerName.normalised("PLAINTEXT"))
+      s"localhost:$port"
+    }.headOption.mkString(",")
+  }
+
+  def currentLeader(client: JAdminClient, topicPartition: TopicPartition): Option[Int] = {
+    Option(
+      client
+        .describeTopics(List(topicPartition.topic).asJava)
+        .all
+        .get
+        .get(topicPartition.topic)
+        .partitions
+        .get(topicPartition.partition)
+        .leader
+    ).map(_.id)
+  }
+
+  def waitForLeaderToBecome(client: JAdminClient, topicPartition: TopicPartition, leader: Option[Int]): Unit = {
+    TestUtils.waitUntilTrue(
+      () => currentLeader(client, topicPartition) == leader,
+      s"Expected leader to become $leader", 10000
+    )
+  }
+
+  def waitForBrokerOutOfIsr(client: JAdminClient, partitions: Set[TopicPartition], brokerId: Int): Unit = {
+    TestUtils.waitUntilTrue(
+      () => {
+        val description = client.describeTopics(partitions.map(_.topic).asJava).all.get.asScala
+        val isr = description.values.flatMap(_.partitions.asScala.flatMap(_.isr.asScala))
+        isr.forall(_.id != brokerId)
+      },
+      s"Expect broker $brokerId to no longer be in any ISR for $partitions"
+    )
+  }
+
+  def waitForBrokerInIsr(client: JAdminClient, partitions: Set[TopicPartition], brokerId: Int): Unit = {
+    TestUtils.waitUntilTrue(
+      () => {
+        val description = client.describeTopics(partitions.map(_.topic).asJava).all.get.asScala
+        val isr = description.values.flatMap(_.partitions.asScala.flatMap(_.isr.asScala))
+        isr.exists(_.id != brokerId)
```

Comment:
should this be == ?

Reply:
Yep. Good catch. I guess the test never passed. Hmm.